pkg/logging/logging.go

```diff
@@ -167,86 +153,93 @@ func init() {
 type LogOptions map[string]string
 
 // InitializeDefaultLogger returns a logrus Logger with a custom text formatter.
-func InitializeDefaultLogger() *logrus.Logger {
-	logger := logrus.New()
-	logger.Formatter = GetFormatter(DefaultLogFormat)
-	logger.SetLevel(LevelStringToLogrusLevel[DefaultLogLevelStr])
-	return logger
-}
-
-// GetLogLevelFromConfig returns the log level provided via global
-// configuration. If the logging level is invalid, ok will be false.
-func GetLogLevelFromConfig() (logrus.Level, bool) {
-	return logOptions.GetLogLevel()
+func InitializeDefaultLogger() (logger *logrus.Logger) {
+	logger = logrus.New()
+	logger.SetFormatter(GetFormatter(DefaultLogFormat))
+	logger.SetLevel(DefaultLogLevel)
+	return
 }
 
 // GetLogLevel returns the log level specified in the provided LogOptions. If
-// it is not set in the options, ok will be false.
-func (o LogOptions) GetLogLevel() (level logrus.Level, ok bool) {
-	level, ok = LevelStringToLogrusLevel[strings.ToLower(o[LevelOpt])]
+// it is not set in the options, it will return the default level.
+func (o LogOptions) GetLogLevel() (level logrus.Level) {
+	levelOpt, ok := o[LevelOpt]
+	if !ok {
+		return DefaultLogLevel
+	}
+
+	var err error
+	if level, err = logrus.ParseLevel(levelOpt); err != nil {
+		logrus.WithError(err).Error("parsing log level, using default one")
+		return DefaultLogLevel
+	}
+
 	return
 }
 
 // GetLogFormat returns the log format specified in the provided LogOptions. If
-// it is not set in the options or is invalid, ok will be false.
+// it is not set in the options or is invalid, it will return the default format.
 func (o LogOptions) GetLogFormat() LogFormat {
 	formatOpt, ok := o[FormatOpt]
 	if !ok {
 		return DefaultLogFormat
 	}
 
+	formatOpt = strings.ToLower(formatOpt)
 	re := regexp.MustCompile(`^(text|json)$`)
 	if !re.MatchString(formatOpt) {
-		logrus.Errorf("incorrect log format configured '%s', expected 'text' or 'json', defaulting to '%s'", formatOpt, DefaultLogFormat)
+		logrus.WithError(
+			fmt.Errorf("incorrect log format configured '%s', expected 'text' or 'json'", formatOpt),
+		).Error("parsing log format, using default one")
```

Comment:
nit, if we're ignoring the problem and continuing then "Error" is probably too severe. We can use warning instead. Also the message of the warning could be a bit clearer:
```suggestion
		logrus.WithError(
			fmt.Errorf("incorrect log format configured '%s', expected 'text' or 'json'", formatOpt),
		).Warning("Ignoring user-configured log format")
```

Reply:
added as part of 4bdca22ba4aae67f176e34e3863a93e7223307c7 :+1: